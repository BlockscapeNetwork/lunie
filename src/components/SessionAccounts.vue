<template>
  <SessionFrame>
    <div class="session">
      <div class="session-container">
        <div class="session-header">
          <h2 class="session-title">My accounts</h2>
          <p>
            You can use the accounts below to explore Lunie.io and to approve
            transactions.
          </p>
        </div>
        <div class="session-main">
          <AccountList
            :accounts="accounts"
            :button-action="goToLunie"
            button-text="Go to Lunie"
          />
        </div>
        <div class="session-footer">
          <router-link to="/welcome">Want to add another account?</router-link>
        </div>
      </div>
    </div>
  </SessionFrame>
</template>

<script>
import AccountList from 'common/AccountList'
import SessionFrame from 'common/SessionFrame'
export default {
  name: `session-accounts`,
  components: {
    AccountList,
    SessionFrame
  },
  computed: {
    accounts() {
      return this.$store.state.accounts
    }
  },
  methods: {
    goToWelcome() {
      this.$router.push(`/welcome`)
    },
    goToLunie() {
      window.open(
        'https://lunie.io/#/extension',
        '_blank',
        'noreferrer noopener'
      )
    }
  }
}
</script>

<style scoped>
.button-add-account {
  display: flex;
  justify-content: flex-end;
  padding: 1.5rem 0 1rem;
  flex-grow: 1;
  align-self: flex-end;
}
</style>
