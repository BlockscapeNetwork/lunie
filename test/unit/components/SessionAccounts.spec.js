import { shallowMount } from '@vue/test-utils'
import SessionAccounts from '../../../src/components/SessionAccounts.vue'

describe(`SessionAccounts`, () => {
  let $store, wrapper

  beforeEach(() => {
    const accounts = [
      { name: `accountname1`, address: 'cosmos1' },
      { name: `accountname2`, address: 'cosmos2' },
      { name: `accountname3`, address: 'cosmos3' }
    ]
    const getters = {
      session: {
        insecureMode: true,
        browserWithLedgerSupport: null
      },
      lastPage: `/`
    }
    $store = {
      getters,
      commit: jest.fn(),
      dispatch: jest.fn(),
      state: {
        accounts
      }
    }
    wrapper = shallowMount(SessionAccounts, {
      mocks: {
        $store
      },
      stubs: [`router-link`]
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has correct go to lunie method`, () => {
    const openFn = jest.fn()
    window.open = openFn
    wrapper.vm.goToLunie()
    expect(openFn).toHaveBeenCalledWith(
      'https://lunie.io/#/extension',
      '_blank',
      'noreferrer noopener'
    )
  })
})
