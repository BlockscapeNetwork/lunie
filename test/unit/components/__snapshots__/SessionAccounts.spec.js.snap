// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionAccounts has the expected html structure 1`] = `
<sessionframe-stub>
  <div
    class="session"
  >
    <div
      class="session-container"
    >
      <div
        class="session-header"
      >
        <h2
          class="session-title"
        >
          My accounts
        </h2>
         
        <p>
          
          You can use the accounts below to explore Lunie.io and to approve
          transactions.
        
        </p>
      </div>
       
      <div
        class="session-main"
      >
        <accountlist-stub
          accounts="[object Object],[object Object],[object Object]"
          buttonaction="function () { [native code] }"
          buttontext="Go to Lunie"
        />
      </div>
       
      <div
        class="session-footer"
      >
        <router-link-stub
          to="/welcome"
        >
          Want to add another account?
        </router-link-stub>
      </div>
    </div>
  </div>
</sessionframe-stub>
`;
